hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/throttling/DeterministicThrottlingTest.java

```diff
@@ -1285,6 +1285,55 @@ void constructsExpectedBucketsFromTestResource() throws IOException {
         assertEquals(expected, rebuilt);
     }
 
+    @Test
+    void canTakeSnapshots() throws IOException {
+        // setup:
+        var defs = SerdeUtils.pojoDefs("bootstrap/throttles.json");
+
+        // and:
+        var expected = List.of(
+                DeterministicThrottle.withMtpsAndBurstPeriod(15_000_000, 2),
+                DeterministicThrottle.withMtpsAndBurstPeriod(5_000, 2),
+                DeterministicThrottle.withMtpsAndBurstPeriod(50_000, 3),
+                DeterministicThrottle.withMtpsAndBurstPeriod(5000, 4));
+        useSomeFromEach(expected);
```

Comment:
I would _almost_ call this `deterministicUseSomeFromEach` (or something similar) since it is critical here (and in subsequent test) that it is in fact deterministic and that isn't necessarily clear from the name "use some".  And least it wasn't to me, until I looked.